### ISSUE TITLE
Column not found: 1054 Unknown column 'download_count' in 'field list' (SQL: update `system_files` set `download_count` = `download_count` + 1,...

### ISSUE THREAD

**Issue by dathwa** [OFFLINE-GmbH/oc-mall-plugin#1137]
Hello.
Mall 3.2.1 and getting error below.
Seems to be an issue on line 61 in VirtualProductFileDownload.php
This is a fle that was generated "on-the-fly".
There is of course no download_count in system_filess.
For now I have commented line 61.

`SQLSTATE[42S22]: Column not found: 1054 Unknown column 'download_count' in 'field list' (SQL: update 'system_files' set 'download_count' = 'download_count' + 1, 'system_files'.'updated_at' = 2024-05-29 07:20:37 where 'id' = 39800 order by 'system_files'.'sort_order' asc)`

```
#0 /vendor/laravel/framework/src/Illuminate/Database/Connection.php(720): Illuminate\Database\Connection->runQueryCallback('update `system_...', Array, Object(Closure))
#1 /vendor/laravel/framework/src/Illuminate/Database/Connection.php(577): Illuminate\Database\Connection->run('update `system_...', Array, Object(Closure))
#2 /vendor/laravel/framework/src/Illuminate/Database/Connection.php(510): Illuminate\Database\Connection->affectingStatement('update `system_...', Array)
#3 /vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(3357): Illuminate\Database\Connection->update('update `system_...', Array)
#4 /vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(3487): Illuminate\Database\Query\Builder->update(Array)
#5 /vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(3463): Illuminate\Database\Query\Builder->incrementEach(Array, Array)
#6 /vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(1077): Illuminate\Database\Query\Builder->increment('download_count', 1, Array)
#7 /vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(969): Illuminate\Database\Eloquent\Builder->increment('download_count', 1, Array)
#8 /vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(926): Illuminate\Database\Eloquent\Model->incrementOrDecrement('download_count', 1, Array, 'increment')
#9 /vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(2318): Illuminate\Database\Eloquent\Model->increment('download_count')
#10 /vendor/october/rain/src/Extension/ExtendableTrait.php(509): Illuminate\Database\Eloquent\Model->__call('increment', Array)
#11 /vendor/october/rain/src/Database/Model.php(293): October\Rain\Database\Model->extendableCall('increment', Array)
#12 /plugins/offline/mall/classes/downloads/VirtualProductFileDownload.php(61): October\Rain\Database\Model->__call('increment', Array)
#13 /vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(46): OFFLINE\Mall\Classes\Downloads\VirtualProductFileDownload->handle('Z3gxckc3U0p1N05...', NULL)
#14 /vendor/laravel/framework/src/Illuminate/Routing/Route.php(260): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(OFFLINE\Mall\Classes\Downloads\VirtualProductFileDownload), 'handle')
#15 /vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\Routing\Route->runController()
#16 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(798): Illuminate\Routing\Route->run()
#17 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#18 /vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#19 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#20 /vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#21 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#22 /vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#23 /vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\Session\Middleware\StartSession->handleStatefulRequest(Object(Illuminate\Http\Request), Object(Illuminate\Session\Store), Object(Closure))
#24 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#25 /vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#26 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#27 /vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#28 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#29 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#30 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(799): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#31 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(776): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#32 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(740): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#33 /vendor/october/rain/src/Router/CoreRouter.php(32): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#34 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(190): October\Rain\Router\CoreRouter->dispatch(Object(Illuminate\Http\Request))
#35 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#36 /plugins/offline/speedy/classes/middleware/CDNMiddleware.php(23): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#37 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): OFFLINE\Speedy\Classes\Middleware\CDNMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#38 /plugins/offline/speedy/classes/middleware/Http2Middleware.php(38): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#39 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): OFFLINE\Speedy\Classes\Middleware\Http2Middleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#40 /plugins/offline/responsiveimages/classes/ResponsiveImagesMiddleware.php(30): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#41 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): OFFLINE\ResponsiveImages\Classes\ResponsiveImagesMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#42 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#43 /vendor/october/rain/src/Foundation/Http/Middleware/CheckForMaintenanceMode.php(23): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle(Object(Illuminate\Http\Request), Object(Closure))
#44 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): October\Rain\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#45 /plugins/vdlp/redirect/classes/RedirectMiddleware.php(78): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#46 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(180): Vdlp\Redirect\Classes\RedirectMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#47 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(116): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#48 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(165): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#49 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(134): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#50 /index.php(43): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#51 {main}
```

**Comment by dathwa** [comment]
With the comment on line 61, It is consequently also breaking on line 64
`$filename = sprintf('%s.%s', $file->display_name, $file->file->getExtension());`

**Comment by tobias-kuendig** [comment]
This is fixed in v3.3.0.
